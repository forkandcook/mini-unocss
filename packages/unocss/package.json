{
  "name": "unocss",
  "version": "0.1.0-beta.2",
  "description": "",
  "keywords": [],
  "homepage": "https://github.com/antfu/unocss#readme",
  "bugs": {
    "url": "https://github.com/antfu/unocss/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/antfu/unocss.git"
  },
  "funding": "https://github.com/sponsors/antfu",
  "license": "MIT",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    },
    "./vite": {
      "require": "./dist/vite.js",
      "import": "./dist/vite.mjs"
    }
  },
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "index.d.ts",
  "files": [
    "dist",
    "*.d.ts"
  ],
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch src",
    "prepublishOnly": "nr build"
  },
  "dependencies": {
    "@unocss/core": "workspace:*",
    "@unocss/preset-attributify": "workspace:*",
    "@unocss/preset-uno": "workspace:*",
    "@unocss/vite": "workspace:*"
  },
  "devDependencies": {
    "nanoevents": "^6.0.2"
  }
}
