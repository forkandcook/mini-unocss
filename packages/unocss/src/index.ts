export * from '@unocss/core'
export { default as presetUno } from '@unocss/preset-uno'
export { default as presetAttributify } from '@unocss/preset-attributify'
