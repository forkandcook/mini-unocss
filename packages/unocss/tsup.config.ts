import { Options } from 'tsup'

const config: Options = {
  splitting: true,
  format: ['esm', 'cjs'],
  entryPoints: [
    'src/*.ts',
  ],
  clean: true,
  dts: true,
}

export default config
