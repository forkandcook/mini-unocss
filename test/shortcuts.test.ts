import { createGenerator, presetUno } from 'unocss'

describe('shortcuts', () => {
  const uno = createGenerator({
    shortcuts: [
      {
        sh1: 'px-2 py-3',
        sh2: 'hover:text-lg text-sm text-lg',
        sh3: 'sm:m-2 m-3',
      },
      [/^button-(\d)$/, ([, d]) => [`px${(+d) * 3}`, `py${(+d) * 2}`]],
    ],
    presets: [
      presetUno(),
    ],
  })

  test('static', async() => {
    const { css } = await uno.generate('sh1 sh2 focus:sh2 sh3')
    expect(css).toMatchSnapshot()
  })

  test('dynamic', async() => {
    const { css } = await uno.generate('button-1 button-2')
    expect(css).toMatchSnapshot()
  })
})
