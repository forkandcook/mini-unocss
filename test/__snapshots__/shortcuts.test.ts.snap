// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shortcuts dynamic 1`] = `
".button-1{padding-left:0.75rem;padding-right:0.75rem;padding-top:0.5rem;padding-bottom:0.5rem;}
.button-2{padding-left:1.5rem;padding-right:1.5rem;padding-top:1rem;padding-bottom:1rem;}"
`;

exports[`shortcuts static 1`] = `
".sh1{padding-left:0.5rem;padding-right:0.5rem;padding-top:0.75rem;padding-bottom:0.75rem;}
.sh3{margin:0.75rem;}
.focus\\\\:sh2:focus,.sh2{font-size:0.875rem;line-height:1.25rem;font-size:1.125rem;line-height:1.75rem;}
.focus\\\\:sh2:hover:focus,.sh2:hover{font-size:1.125rem;line-height:1.75rem;}
@media (min-width: 640px){
.sh3{margin:0.5rem;}
}"
`;
